tests/test_healthsystem.py

```diff
@@ -1952,6 +1956,128 @@ def apply(self, person_id, squeeze_factor):
+    keys_district = list(person_for_district.keys())
+
+    for i in range(0, int(tot_population)):
+        sim.population.props.at[i, 'district_of_residence'] = keys_district[0]
+
+
+    # Schedule an identical appointment for all individuals
+    for i in range(0, tot_population):
+
+        hsi = DummyHSIEvent(module=sim.modules['DummyModule'],
+                            person_id=i,
+                            appt_type='MinorSurg',
+                            level='1a')
+
+        sim.modules['HealthSystem'].schedule_hsi_event(
+            hsi,
+            topen=sim.date,
+            tclose=sim.date + pd.DateOffset(days=1),
+            # Assign equal priority
+            priority=0
+        )
+
+    hsi1 = DummyHSIEvent(module=sim.modules['DummyModule'],
+                         person_id=0,  # Ensures call is on officers in first district
+                         appt_type='MinorSurg',
+                         level='1a')
+    hsi1.initialise()
+    
+    # Get facility ID
+    facID = int((re.search(r'\d+', next(iter(hsi1.expected_time_requests)))).group())
+
+    pharmacy_task_time = hsi1.expected_time_requests['FacilityID_' + str(facID) + '_Officer_Pharmacy']
+    nursing_task_time = hsi1.expected_time_requests['FacilityID_' + str(facID) + '_Officer_Nursing_and_Midwifery']
+    clinical_task_time = hsi1.expected_time_requests['FacilityID_' + str(facID) + '_Officer_Clinical']
+ 
+     # Check that first choice of task-shifting officer for Pharmacy is Nursing_and_Midwifery, and get their factor
+    assert 'Nursing_and_Midwifery' == sim.modules['HealthSystem'].global_task_shifting.get('Pharmacy')[0][0]
+    nursing_task_shift_factor = sim.modules['HealthSystem'].global_task_shifting.get('Pharmacy')[1][0]
+ 
+    # Number of appts that want to see delivered if pharmacy time is set to zero,
+    # clinical time is set to perform 50 appts, and nursing time is set to perform 50 appts including
+    # both nursing and pharmacy tasks.
+    Ntarget = 50
+    
+    sim.modules['HealthSystem']._daily_capabilities['FacilityID_' + str(facID) + '_Officer_Pharmacy'] = 0.0
+    sim.modules['HealthSystem']._daily_capabilities['FacilityID_' + str(facID) + '_Officer_Clinical'] = Ntarget*(clinical_task_time)
+    sim.modules['HealthSystem']._daily_capabilities['FacilityID_' + str(facID) + '_Officer_Nursing_and_Midwifery'] = Ntarget*(nursing_task_time + nursing_task_shift_factor*pharmacy_task_time)
+
+    # Run healthsystemscheduler
+    healthsystemscheduler.apply(sim.population)
+
+    # read the results
+    output = parse_log_file(sim.log_filepath, level=logging.DEBUG)
+    hs_output = output['tlo.methods.healthsystem']['HSI_Event']
+
+    # Check that all events could run, even if Pharmacy capabilities were set to zero,
+    # and that when task-shifting first option (nurses) where always preferentially chosen over second
+    # one (clinicians)
+    assert hs_output['did_run'].sum() == Ntarget
+
```

Comment:
To strengthen the test and see the impact of task-shifting versus no task-shifting, I think it would be good to repeat this for with/without global task-shifting so we can see the difference.